eva-accession-deprecate/src/main/java/uk/ac/ebi/eva/accession/deprecate/configuration/ClusteredVariantsToDeprecateReaderConfiguration.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.deprecate.configuration;
+
+import com.mongodb.MongoClient;
+import org.springframework.batch.core.configuration.annotation.StepScope;
+import org.springframework.boot.autoconfigure.mongo.MongoProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+import uk.ac.ebi.eva.accession.core.configuration.MongoConfiguration;
+import uk.ac.ebi.eva.accession.deprecate.io.ClusteredVariantsToDeprecateReader;
+
+@Configuration
+@Import({MongoConfiguration.class})
+public class ClusteredVariantsToDeprecateReaderConfiguration {
+
+    @Bean
+    @StepScope
+    ClusteredVariantsToDeprecateReader clusteredVariantsToDeprecateReader(MongoClient mongoClient,
+                                                                          MongoProperties mongoProperties) {
```

Comment:
Since we are using only the database name, why pass in the full MongoProperties object? Let's just pass in the database name.

Reply:
As we are using `MongoConfiguration` we already have a Bean for `MongoProperties`. It will read from the properties file the parameters with the prefix `spring.data.mongodb` and the database name will be there. If we pass the String for the database we will need to define a Bean for it and read from the properties file.

I would prefer use the `MongoProperties` object